aaelf32/aaelf32.rst

```diff
@@ -0,0 +1,3289 @@
+this specification.
+
+Idempotency
+^^^^^^^^^^^
+
+All RELA type relocations are idempotent.  They may be reapplied to the place
+and the result will be the same.  This allows a static linker to preserve full
+relocation information for an image by converting all REL type relocations into
+RELA type relocations.
+
+.. note::
+   A REL type relocation can never be idempotent because the act of applying
+   the relocation destroys the original addend.
+
+Program Loading and Dynamic Linking
+===================================
+
+This section provides details of Arm-specific definitions and changes relating
+to executable images.
+
+Program Header
+--------------
+
+The Program Header provides a number of fields that assist in interpretation of
+the file.  Most of these are specified in the base standard.  The following
+fields have Arm-specific meanings.
+
+.. rubric:: p_type
+
+The following table lists the processor-specific segment types.
+
+.. table:: Processor-specific segment types
+
+  +-----------------+------------+-------------------------------------------------+
+  | Name            | p_type     | Meaning                                         |
+  +=================+============+=================================================+
+  | PT_ARM_ARCHEXT  | 0x70000000 | Platform architecture compatibility information |
+  +-----------------+------------+-------------------------------------------------+
+  | | PT_ARM_EXIDX  | 0x70000001 | Exception unwind tables                         |
+  | | PT_ARM_UNWIND |            |                                                 |
+  +-----------------+------------+-------------------------------------------------+
+
+A segment of type PT_ARM_ARCHEXT contains information describing the platform
+capabilities required by the executable file.  The segment is optional, but if
+present it must appear before segment of type PT_LOAD.  The platform
+independent parts of this segment are described in `Platform architecture compatibility data`_.
+
+PT_ARM_EXIDX (alias PT_ARM_UNWIND) describes the location of a program’s unwind
+tables.
+
+.. rubric:: p_flags
+
+There are no processor-specific flags.  All bits in the PT_MASKPROC part of
+this field are reserved to future revisions of this specification.
+
+Platform architecture compatibility data
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This data describes the platform capabilities required by an executable file.
+It can be constructed by a linker using the attributes [Addenda32_] found in
+its input relocatable files, or otherwise.
+
+If this segment is present it shall contain at least one 32-bit word with
+meaning defined by `Common architecture compatibility data masks`_,
+`Architecture compatibility data formats`_, `Architecture profile compatibility
+data`_, and `Architecture version compatibility data`_ below.
+
+.. _Common architecture compatibility data masks:
+
+.. table:: Common architecture compatibility data masks
+  :widths: 35 15 50
```

Comment:
as described above on the version history table, please use a class to specify the width